### PR TITLE
Create codeql.yml

### PR DESCRIPTION
**Pull request by ThibaudLopez**
CodeQL now supports Kotlin:
https://github.blog/changelog/2022-11-28-codeql-code-scanning-launches-kotlin-analysis-support-beta/

I enabled CodeQL using the default YAML that was auto-generated by GitHub CodeQL setup. The default branch was correctly set to `main`. And the CodeQL instructions say to use language `'java'` for Kotlin. The rest of the YAML is default.

### CODE REVIEW

**Comment by tobrun** [comment]
@ThibaudLopez currently the CI fails with:

```
  [2022-12-16 21:08:43] [autobuild] groovy.lang.MissingPropertyException: Could not get unknown property 'SDK_REGISTRY_TOKEN' for root project 'Mapbox Maps Android' of type org.gradle.api.Project.
```

We will need to setup a secret env variable, that allows to download our binaries from SDK registry. More general information about it [here](https://docs.mapbox.com/android/maps/guides/install/#configure-credentials).

**Comment by ThibaudLopez** [comment]
Thanks @tobrun . I see the environment [setup](https://github.com/mapbox/mapbox-maps-android/settings/environments). I added the `environment: env` line. And I signed the CLA.

**Review by tobrun — APPROVED** [review]

**Review by jush — APPROVED** [review]

**Review by pengdev — APPROVED** [review]

**Comment by yunikkk** [comment]
Is it actually processing Kotlin code? In the [report](https://github.com/mapbox/mapbox-maps-android/actions/runs/3836638325/jobs/6531027322) I see
```

|                  Metric                  | Value |
+------------------------------------------+-------+
| Total lines of Java code in the database |   299 |
```
Seems only Java is handled, not Kotlin

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main", v*.* ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
```
[the remark this answers is not in the document]
Comment (ThibaudLopez) [reply .github/workflows/codeql.yml:20]: The YAML was generated automatically by GitHub when setting up CodeQL with default values. The important values to verify are `branches` and `languages`. Are those good?

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    environment: env
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'java' ]
```
[the remark this answers is not in the document]
Comment (ThibaudLopez) [reply .github/workflows/codeql.yml]: I added `python` just now

File: .github/workflows/codeql.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexander-kulikovskii) [reply .github/workflows/codeql.yml:20]: `branches` and `languages` are good for me. 
But there is no need to run any cron task in "silent" mode.
But not super critical

**Review by alexander-kulikovskii — APPROVED** [review]

**Comment by ThibaudLopez** [comment]
I can't merge because the CircleCI build checks are failing.

**Comment by tobrun** [comment]
> It's not possible to merge because the CircleCI build checks are failing. Help?

this is because of the recent incident. Can you login and out of Circle-CI and follow this project. Rebase/push should trigger the jobs again.

**Comment by yunikkk** [comment]
@ThibaudLopez @tobrun any ideas why only 299 lines of Java code is reported

```
|                  Metric                  | Value |
+------------------------------------------+-------+
| Total lines of Java code in the database |   299 |
``` 
in the job https://github.com/mapbox/mapbox-maps-android/actions/runs/3876146401/jobs/6609623519 ?

**Comment by ThibaudLopez** [comment]
> Can you login and out of Circle-CI and follow this project. Rebase/push should trigger the jobs again.

@tobrun, OK, I'm re-trying... I re-authorized the CircleCI app to my account, and I pushed an empty commit.

---

> any ideas why only 299 lines of Java code is reported

@yunikkk, I guess we'll have to wait for GitHub to get Kotlin out of beta, or find help somewhere.

**Comment by ThibaudLopez** [comment]
@tobrun , @yunikkk , can we merge this PR? The CodeQL build checks pass. The remaining errors seem unrelated to CodeQL. As for the remaining problem of only reporting on 299 lines, let's merge now and start as is, and let's get further help from GitHub Support once they get Kotlin out of beta.

**Comment by yunikkk** [comment]
> @tobrun , @yunikkk , can we merge this PR? The CodeQL build checks pass. The remaining errors seem unrelated to CodeQL. As for the remaining problem of only reporting on 299 lines, I propose we merge now and start as is, and get further help from GitHub Support once they get Kotlin out of beta.

I'm good with it, would be interesting to track when Kotlin will be analyzed actually.

**Review by yunikkk — APPROVED** [review]

**Comment by ThibaudLopez** [comment]
see https://github.com/mapbox/mapbox-navigation-android/pull/7048 for example of CodeQL setup for Kotlin

**Comment by yunikkk** [comment]
> see [mapbox/mapbox-navigation-android#7048](https://github.com/mapbox/mapbox-navigation-android/pull/7048) for example of CodeQL setup for Kotlin

So the trick is to call [make command](https://github.com/mapbox/mapbox-navigation-android/actions/runs/5200829092/workflow?pr=7048#L40) explicitly instead of relying of autobuild?

**Pull request merged**